tests/featureTesting/compare_json.py

```diff
@@ -0,0 +1,149 @@
+
+import helper_functions as hf
+
+import pandas as pd
+
+
+class CompareJsons(object):
+    def __init__(self, man_path, auto_path, analysis_type = cs.by_delta_rt_and_mz, n_rt=5,
+                 n_mz = 2,delta_rt = 0.2, delta_mz =0.3,corr_cutoff = 1, p_val_cutoff = 0.7,
+                 comparater_list= None,unique_identifier_list= None):
+        if unique_identifier_list is None:
+            unique_identifier_list = []
+        if comparater_list is None:
+            comparater_list = []
+
+        self.man_path = man_path
+        self.auto_path = auto_path
+        self.analysis_type = analysis_type
+        self.n_rt = n_rt
+        self.n_mz = n_mz
+        self.delta_rt = delta_rt
+        self.delta_mz = delta_mz
+        self.corr_cutoff = corr_cutoff
+        self.p_val_cutoff = p_val_cutoff
+        self.comparater_list = comparater_list
+        self.unique_identifier_list = unique_identifier_list
+
+
+
+    def summary_report(self):
+        correlation_and_wilcox_df = self.correlation_and_wilcox()
+        correlation_plot_url = self.correlation_plot()
+
+
+
+
+
+
+    def correlation_and_wilcox(self):
+        comp_dict_man = hf.json_comp_data(self.man_path)
+        comp_dict_auto = hf.json_comp_data(self.auto_path)
+        if self.analysis_type == cs.by_delta_rt_and_mz:
+            correlation_and_wilcox_df= hf.find_correlation_and_wilcox_test_df\
+                (self.unique_identifier_list, comp_dict_man, comp_dict_auto, self.comparater_list,
+                 self.analysis_type, self.delta_rt, self.delta_mz)
+
+        elif self.analysis_type == cs.by_n_closest_rt_and_mz:
+            correlation_and_wilcox_df = hf.find_correlation_and_wilcox_test_df_closest_rt_and_mz\
+                (self.unique_identifier_list, comp_dict_man, comp_dict_auto, self.comparater_list,
+                 self.n_rt, self.n_mz, self.delta_rt, self.delta_mz, self.analysis_type)
+
+        elif self.analysis_type == cs.by_closest_and_delta_rt_and_mz:
+            correlation_and_wilcox_df = hf.find_correlation_and_wilcox_test_df_closest_rt_and_mz\
+                (self.unique_identifier_list,comp_dict_man, comp_dict_auto, self.comparater_list,
+                 self.n_rt, self.n_mz,self.delta_rt,self.delta_mz,self.analysis_type)
+        return correlation_and_wilcox_df
+
+
+    def correlation_plot(self):
+        correlation_and_wilcox_df = self.correlation_and_wilcox()
+        correlation_df = correlation_and_wilcox_df[[cs.unique_identifier_man,
+                                                    cs.unique_identifier_auto, cs.corr_coff,
+                                                   cs.avg_intensity_man, cs.avg_intensity_auto]]
+        config_name = hf.get_basename_url(self.man_path) + hf.get_basename_url(self.auto_path)
+        cor_plot = hf.get_correlation_plot(correlation_df,config_name)
+        return cor_plot
+
+
+    def wilcox_plot(self):
+        correlation_and_wilcox_df = self.correlation_and_wilcox()
```

Comment:
Make sure you write documentation for all the function and classes you have defined.

Reply:
@harshitcodes : Done